src/main/java/seedu/taassist/model/student/Student.java

```diff
@@ -54,11 +54,26 @@ public Address getAddress() {
     }
 
     /**
-     * Returns an immutable moduleClass set, which throws {@code UnsupportedOperationException}
-     * if modification is attempted.
+     * Returns an immutable moduleData set as an Unmodifiable ObservableList.
      */
-    public Set<ModuleClass> getModuleClasses() {
-        return Collections.unmodifiableSet(moduleClasses);
+    public List<StudentModuleData> getModuleData() {
+        return moduleData.asUnmodifiableObservableList();
+    }
+
+    /**
+     * Returns a list of module classes that the student is enrolled in.
+     */
+    public List<ModuleClass> getModuleClasses() {
+        return getModuleData().stream()
+            .map(StudentModuleData::getModuleClass)
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the {@code StudentModuleData} of the student for the given {@code ModuleClass}.
+     */
+    public StudentModuleData findStudentModuleData(ModuleClass focusedClass) {
```

Comment:
```suggestion
    public StudentModuleData findStudentModuleData(ModuleClass targetClass) {
```

Perhaps just name it this for abstraction reasons?